### ISSUE TITLE
Error : Action "Iframe" not found in the module "API"

### ISSUE THREAD

**Issue by Nehagaj** [piwik/piwik#18041]
When exporting in an iframe a particular page using the Widgetize feature, when I try to export a report's data, and click on "Export this dataset in other formats" it opens a new window with the error : Error : Action "Iframe" not found in the module "API". The issue is coming up for the list tab in the iframe. It is working fine for the Charts tab. 
What I do not understand is the same action iframe works for the module widgetize for the Charts tab and the same action iframe isnt found in the module widgetize for the List tab.

Can you please let us know how we can rectify this issue?

## Steps to Reproduce (for Bugs)

1.
2.

## Your Environment

* Matomo Version: 
* PHP Version:
* Server Operating System:
* Additionally installed plugins:

* Browser:
* Operating System:

**Comment by sgiehl** [comment]
Hi @Nehagaj 
Could you please provide some simple steps how to reproduce that? If possible you could even provide some example urls or urls from our demo where you are able to produce the error.
Once we are able to reproduce we will see if / when we will be able to fix it.

**Comment by Nehagaj** [comment]
We won't be able to reproduce this issue on demo site of matomo since we are facing this issue in an iFrame that we have embedded into our application.
We are using below url for accessing the stats report : 
https://intranet-piwik.abc/index.php?token_auth=XXXXXXXXXX&module=Widgetize&action=iframe&moduleToWidgetize=Actions&actionToWidgetize=getPageUrls&idSite=1&fllat=1&filter_limit=100&period=week&date=yesterday&segment=pageUrl=@https://www.abc.com/index.php?date=yesterday&expanded=1&filter_limit=100&format=TSV&idSite=1&language=fr&method=Actions.getPageUrls&module=API&period=week&segment=pageUrl%3D@https:%2F%2Fwww.abc.com%2Ftest%2Fxxxxxx%2F&token_auth=xxxxxxxxxxxxx&translateColumnNames=1
which is giving us the following error
![Matomo_issue](https://user-images.githubusercontent.com/91129098/134509127-edceb72c-6236-4ca1-81c0-f8a9418299ca.JPG)

We select the stats report for a particular page and are trying to export the data in following format. refer attached image for the same. Once we click on export we are getting above error.
![Matomo_issue2](https://user-images.githubusercontent.com/91129098/134510462-64774539-5101-442c-93bc-e8fe460371e4.JPG)

**Comment by tsteur** [comment]
@Nehagaj what does the URL look like for the widget that you embed? The URL you mentioned above looks like it wasn't generated correctly (eg `segment` parameter not escaped, parameters there twice etc and two question marks etc). I understand above URL was generated from the exporter?

And you have a different URL for embedding the page URL report into a frame. What's the URL you use for that?

**Comment by Nehagaj** [comment]
Hello,

We use below url. It works fine for displaying the data into iframe. But it is not working while we try to export this data.

url = 'https://intranet-piwik.abc/index.php?token_auth=xyz&apiMethod=Actions.getPageUrls&module=Widgetize&action=iframe&moduleToWidgetize=CoreHome&actionToWidgetize=getRowEvolutionPopover&idSite=1&date=yesterday&period='+periode+'&disableLink=0&widget=1&label='+page_url+'&language=#langueUser#';

Matomo version that we are using : 4.3.1

**Comment by sgiehl** [comment]
@Nehagaj So do you mean the `Export icon` link in the embedded rowevolution report does not work?

**Comment by Nehagaj** [comment]
Yes this is not working. Once we click on export we are not able to generate report and are getting the above error in a new window.

**Comment by Nehagaj** [comment]
We are getting this issue "Action iFrame not found in module API"

**Comment by Nehagaj** [comment]
Any updates on this issue?

**Comment by sgiehl** [comment]
Hi @Nehagaj. I'm actually not able to reproduce this. Using the URL you provided above I'm able to open the export popover and clicking export correctly opens the exported data in a new window. This works correctly for me in Matomo 4.3.1, 4.6.2 and also with latest releases and development version. Maybe updating to the latest version helps.

**Comment by Nehagaj** [comment]
Hello,

We have recently updated the Matomo to version 4.3.1

Also we are getting this issue when we are clicking at the below green export button. Refer screenshot.

<img width="744" alt="export_issue" src="https://user-images.githubusercontent.com/91129098/152124259-ae82ed99-af60-4177-93ba-3696dcb389b0.png">
